### PR TITLE
(SIMP-6197) Add `simp puppetfile generate`

### PR DESCRIPTION
**Pull request by op-ct**
This patch adds the `puppetfile generate` command, which has two modes of
operation:

Generating a `Puppetfile.simp` from the RPM-installed modules local git
repositories:

    simp puppetfile generate > Puppetfile.simp

Scaffolding a site Puppetfile that includes `Puppetfile.simp`:

    simp puppetfile generate --skeleton > Puppetfile

In the course of events, the patch also moved logger to `Simp::Cli::Logging`
and fixed various annoyances that had prevented local smoke tests with
`bin/simp`:
  - Avoid using AIO Puppet with `USE_AIO_PUPPET=no`
  - Load all `simp` commands without `simp config` failing in non-puppetserver
    environments (`simp config` still fails as expected)

SIMP-6193 #close
SIMP-6197 #comment Added `simp puppetfile generate` to simp-cli

### CODE REVIEW

File: lib/simp/cli/puppetfile/skeleton.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        # `Puppetfile.simp` at any time by running the command:
+        #
+        #     simp puppetfile generate > Puppetfile.simp
+        #
+        # You can regenerate a clean copy of this Puppetfile at any time by running:
+        #
+        #     simp puppetfile generate --skeleton > Puppetfile
+        #
+        # ------------------------------------------------------------------------------
+        instance_eval(File.read("Puppetfile.simp"))
```
Comment (trevor-vaughan) [inline lib/simp/cli/puppetfile/skeleton.rb]: Probably should explain this

File: lib/simp/cli/puppetfile/skeleton.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+        # ==============================================================================
+        # Your site's Puppet modules
+        # ------------------------------------------------------------------------------
+        # Add your own Puppet modules here
+
+
+        # ------------------------------------------------------------------------------
```
Comment (trevor-vaughan) [inline lib/simp/cli/puppetfile/skeleton.rb]: I don't really get that this is supposed to be with the previous section (is it?)

File: lib/simp/cli/puppetfile/skeleton.rb (hunk 1 above)
Comment (trevor-vaughan) [reply lib/simp/cli/puppetfile/skeleton.rb]: Also, do we need to check and see if that file exists before loading it? Not sure how much Ruby we want to stuff in this thing.

File: spec/lib/simp/cli/puppetfile/files/simplib.metadata.json [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      "operatingsystemrelease": [
+        "6.1",
+        "7.1"
+      ]
+    }
+  ],
+  "requirements": [
+    {
+      "name": "puppet",
+      "version_requirement": ">= 4.10.4 < 6.0.0"
```
Comment (trevor-vaughan) [inline spec/lib/simp/cli/puppetfile/files/simplib.metadata.json]:
```suggestion
      "version_requirement": ">= 4.10.4 < 7.0.0"
```

File: lib/simp/cli/puppetfile/skeleton.rb (hunk 2 above)
Comment (op-ct) [reply lib/simp/cli/puppetfile/skeleton.rb]: Nope, it's just a separate comment marking where you should put your own stuff (as opposed to the SIMP section above).

File: lib/simp/cli/puppetfile/skeleton.rb (hunk 1 above)
Comment (op-ct) [reply lib/simp/cli/puppetfile/skeleton.rb]: > Probably should explain this

I couldn't think of a concise way to explain the technical implementation of this line for Ruby nerds<sup>*</sup>) in a way that wouldn't also obfuscate the basic purpose of and usage of the file for general users.  This is something we can tackle in a follow-on ticket if we think of a better way.

(* = basically, `require` and `load` won't work while r10k is processing the file, so you need to do this)



> Also, do we need to check and see if that file exists before loading it? Not sure how much Ruby we want to stuff in this thing.

This `Puppetfile` is only generated by `generate --skeleton` in order to use `Puppetfile.simp`—I considered wrapping it in additional logic to spit out a prettier error message, but seemed to add more clutter than it was worth to an already-busy Puppetfile.

**Pull request merged**